app.py

```diff
@@ -1,54 +1,128 @@
-from flask import Flask, render_template, request, redirect, url_for
+from flask import Flask, render_template, request, redirect, url_for, abort
 import db_items as mongoDB
 from viewmodel import ViewModel
-from app_config import Config
+from user import User
 import pymongo
 import certifi
 import os
+import requests
+from flask_login import LoginManager, login_required, login_user, current_user
+from oauthlib.oauth2 import WebApplicationClient
+   
+secret_key = os.environ.get('SECRET_KEY', 'secret_key')
+admin_user = os.environ.get('APP_ADMIN_USER_ID', 'admin')
+client_id = os.getenv('CLIENT_ID')
+client_secret = os.getenv('CLIENT_SECRET')
+login_manager = LoginManager()
+appClient = WebApplicationClient(client_id)
+
+ROLE_READER = 'reader'
+ROLE_WRITER = 'writer'
 
 def create_app():
     app = Flask(__name__) 
-    app.config.from_object(Config())
 
-    board_id = os.getenv('BOARD_ID')
+    login_disabled = os.environ.get('LOGIN_DISABLED', 'False') == 'True'
+    app.config['LOGIN_DISABLED'] = login_disabled
     
+    board_id = os.getenv('BOARD_ID')
     db_connectionstring = os.getenv('MONGO_CONNECTION_URL')
+    
 
     client = pymongo.MongoClient(
         db_connectionstring,
         tlsCAFile=certifi.where()
     )
     db = client.TodoListDB
     collection = db.todos
+
+    @login_manager.unauthorized_handler 
+    def unauthenticated():
+        uri = appClient.prepare_request_uri("https://github.com/login/oauth/authorize")
+        return redirect(uri)
+        
+    @login_manager.user_loader 
+    def load_user(user_id):
+        return User(user_id) 
+        # return None
+
+    @app.route('/login/callback', methods=['GET'])
+    def login_callback():
+        token_response = requests.post('https://github.com/login/oauth/access_token', data={
+                'client_id': client_id,
+                'client_secret': client_secret,
+                'code': request.args.get('code')
+            }, headers={'Accept': 'application/json'})
+
+        appClient.parse_request_body_response(token_response.text)
+
+        user_response = requests.get('https://api.github.com/user', headers={
+            'Accept': 'application/json', 
+            'Authorization': f"token {token_response.json()['access_token']}"
+            })
+
+        login_user(User(user_response.json()['login']))        
+        return redirect(url_for('index'))
```

Comment:
What you've done is fine but instead of knowing the correct requests to make, you could have got the appClient to tell you via appClient.prepare_token_request and `appClient.add_token`